webapp-django/crashstats/supersearch/form_fields.py

```diff
@@ -116,6 +118,12 @@ def value_to_string(self, value):
         except AttributeError:  # when value is None
             return value
 
+    def strptime(self, value, format):
```

Comment:
Hmm... 
Considering that `settings.USE_TZ` is on by default, the regular Django `DateTimeField` will return a timezone aware datetime object. Whereas `isodate.parse_datetime(value)` won't. 
We might just be lucky that things work today, but it's dangerous territory. 
Like mixing unicode and bytestring. 

In general, the winner always; Use UTC, Use Unicode, Use timezone aware dates everywhere. 

The default Django `DateTimeField` returns a timezone aware datetime object, 
```
>>> from django.forms import DateTimeField
>>> d=DateTimeField()
>>> d.clean('2015-09-23 12:13:14')
datetime.datetime(2015, 9, 23, 12, 13, 14, tzinfo=<django.utils.timezone.LocalTimezone object at 0x10b162850>)
```
Strangely, the same test doesn't work with your class:
```
>>> from crashstats.supersearch.form_fields import DateTimeField
>>> d=DateTimeField()
>>> d.clean('2015-09-23 12:13:14')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/supersearch/form_fields.py", line 100, in clean
    **kwargs
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/supersearch/form_fields.py", line 49, in clean
    cleaned_value = super(PrefixedField, self).clean(*args, **kwargs)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/forms/fields.py", line 150, in clean
    value = self.to_python(value)
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/supersearch/form_fields.py", line 46, in to_python
    return super(PrefixedField, self).to_python(value)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/forms/fields.py", line 504, in to_python
    result = super(DateTimeField, self).to_python(value)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/forms/fields.py", line 416, in to_python
    raise ValidationError(self.error_messages['invalid'], code='invalid')
ValidationError: [u'Enter a valid date/time.']
>>> d.clean('2015-09-23T12:13:14')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/supersearch/form_fields.py", line 100, in clean
    **kwargs
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/supersearch/form_fields.py", line 49, in clean
    cleaned_value = super(PrefixedField, self).clean(*args, **kwargs)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/forms/fields.py", line 150, in clean
    value = self.to_python(value)
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/supersearch/form_fields.py", line 46, in to_python
    return super(PrefixedField, self).to_python(value)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/forms/fields.py", line 504, in to_python
    result = super(DateTimeField, self).to_python(value)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/forms/fields.py", line 416, in to_python
    raise ValidationError(self.error_messages['invalid'], code='invalid')
ValidationError: [u'Enter a valid date/time.']
```
So basically, that simple `.clean()` method doesn't work on our custom class at all.